extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -377,4 +316,37 @@ private static RowSignature computeSignatureForFinalWindowStage(RowSignature row
         finalWindowClusterBy.getColumns()
     );
   }
+
+  /**
+   * This method converts the operator chain received from native plan into MSQ plan.
+   * (NaiveSortOperator -> Naive/GlueingPartitioningOperator -> WindowOperator) is converted into (GlueingPartitioningOperator -> PartitionSortOperator -> WindowOperator).
+   * We rely on MSQ's shuffling to do the clustering on partitioning keys for us at every stage.
+   * This conversion allows us to blindly read N rows from input channel and push them into the operator chain, and repeat until the input channel isn't finished.
+   * @param operatorFactoryListFromQuery
+   * @param maxRowsMaterializedInWindow
+   * @return
+   */
+  private List<OperatorFactory> getOperatorFactoryListForStageDefinition(List<OperatorFactory> operatorFactoryListFromQuery, int maxRowsMaterializedInWindow)
+  {
+    final List<OperatorFactory> operatorFactoryList = new ArrayList<>();
+    final List<OperatorFactory> sortOperatorFactoryList = new ArrayList<>();
```

Comment:
wouldn't this logic change the order of operators; why does it want to do that?
```
Sort
Partition
Something1
Sort
Partition
Something2
```
to 
```
Partition
Sort
Something1
Partition
Sort
Something2
```

I think the 1st sort operator could possibly be removed entirely as the shuffle should take care of that.

I believe changing the order of operators should be a planner decision; I don't think that may happen here.
how about:
* trash the 1st sort operator (the shuffle takes care of that)
* leave all further `SortOperator` alone for now - and get back to that later

Reply:
we will improve on this later - for now this removes the need to fully sort all rows